types/node/readline/promises.d.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * readline/promises
+ * @since v17.0.0
+ */
+declare module 'readline/promises' {
+    import { EventEmitter } from "events";
+    import { CursorPos, ReadLineOptions, Direction, CompleterResult, Key } from "readline";
+    interface InterfaceEvents {
+        close: () => void;
+        line: (input: string) => void;
+        history: (history: string[]) => void;
+        pause: () => void;
+        resume: () => void;
+        SIGCONT: () => void;
+        SIGINT: () => void;
+        SIGTSTP: () => void;
+    }
+    type Completer = (line: string) => CompleterResult | Promise<CompleterResult>;
+    class Interface extends EventEmitter {
```

Comment:
According to the docs this extends `readline.Interface`, you can remove most of this class declaration.

Reply:
I tried to extend the callback class and override the different functions, I may have done it wrong but it did not work for me.